x/concentrated-liquidity/types/keys.go

```diff
@@ -1,33 +1,43 @@
 package types
 
 import (
-	"fmt"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
 )
 
 const (
 	ModuleName = "concentrated-liquidity"
 
-	StoreKey  = ModuleName
-	RouterKey = ModuleName
-
-	QuerierRoute = ModuleName
-	// Contract: Coin denoms cannot contain this character
-	KeySeparator = "|"
+	StoreKey = ModuleName
 )
 
+// Key prefixes
 var (
-	TickPrefix     = "tick_prefix" + KeySeparator
-	PositionPrefix = "position_prefix" + KeySeparator
+	TickPrefix     = []byte{0x01}
+	PositionPrefix = []byte{0x02}
 )
 
-// KeyTick uses pool Id and tick index
-func KeyTick(poolId uint64, tickIndex sdk.Int) []byte {
-	return []byte(fmt.Sprintf("%s%d%s", TickPrefix, poolId, tickIndex.String()))
+// KeyTick returns a key for storing a TickInfo object.
+func KeyTick(poolId uint64, tickIndex int64) []byte {
+	key := KeyTickPrefix(poolId)
+	key = append(key, sdk.Uint64ToBigEndian(uint64(tickIndex))...)
+	return key
+}
+
+// KeyTickPrefix constructs a key prefix for storing a TickInfo object.
+func KeyTickPrefix(poolId uint64) []byte {
+	var key []byte
+	key = append(key, TickPrefix...)
+	key = append(key, sdk.Uint64ToBigEndian(poolId)...)
+	return key
 }
 
 // KeyPosition uses pool Id, owner, lower tick and upper tick for keys
-func KeyPosition(poolId uint64, address string, lowerTick, upperTick sdk.Int) []byte {
-	return []byte(fmt.Sprintf("%s%d%s%s%s", PositionPrefix, poolId, address, lowerTick.String(), upperTick.String()))
+func KeyPosition(poolId uint64, addr sdk.AccAddress, lowerTick, upperTick int64) []byte {
+	var key []byte
+	key = append(key, PositionPrefix...)
+	key = append(key, address.MustLengthPrefix(addr)...)
+	key = append(key, sdk.Uint64ToBigEndian(uint64(lowerTick))...)
+	key = append(key, sdk.Uint64ToBigEndian(uint64(upperTick))...)
```

Comment:
we do have negative ticks though right?

Reply:
Yes! int64 vs uint64 are the same values, they just are represented differently.

E.g.

https://go.dev/play/p/irSTd51l4vz